CMakeLists.txt

```diff
@@ -233,6 +235,20 @@ if (NRN_ENABLE_CORENEURON)
 endif()
 
 
+# =============================================================================
+# Add tests if enabled
+# =============================================================================
+if (NRN_ENABLE_TESTS)
+  add_external_project(catch2)
+  set(CATCH_DIR ${PROJECT_SOURCE_DIR}/external/catch2)
```

Comment:
Is it necessary in all cases to build catch2 as a submodule? I generally like the build speed benefits of allowing
external packages to be built separately and then found. Eg. See the NRN_ENABLE_CORENEURON support and NRN_ENABLE_INTERVIEWS sections.

Reply:
catch2 submodule will be only downloaded if `NRN_ENABLE_TESTS` is `ON`. We can keep that `OFF` by default.

By the [catch2](https://github.com/catchorg/Catch2) is just header file so there is nothing to build.